tests/lib.rs

```diff
@@ -3,57 +3,62 @@
 extern crate iso8601;
```

Comment:
Only realizing now that all the parsers stuff is still public.
We should probably move the tests here into becoming unit tests inside the crate and replace it here with proper integration tests of the (smaller) public API. 

I'll do that in a follow-up though.